compiler/rustc_hir_analysis/src/hir_ty_lowering/object_safety.rs

```diff
@@ -86,47 +85,17 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         let (mut auto_traits, regular_traits): (Vec<_>, Vec<_>) =
             expanded_traits.partition(|i| tcx.trait_is_auto(i.trait_ref().def_id()));
         if regular_traits.len() > 1 {
-            let first_trait = &regular_traits[0];
-            let additional_trait = &regular_traits[1];
-            let mut err = struct_span_code_err!(
-                tcx.dcx(),
-                additional_trait.bottom().1,
-                E0225,
-                "only auto traits can be used as additional traits in a trait object"
+            let _ = self.report_trait_object_addition_traits_error(
+                span,
+                &trait_bounds,
+                &regular_traits,
             );
-            additional_trait.label_with_exp_info(
-                &mut err,
-                "additional non-auto trait",
-                "additional use",
+        } else if regular_traits.is_empty() && auto_traits.is_empty() {
+            let reported = self.report_trait_object_addition_traits_error(
```

Comment:
use a different function for this branch instead of reusing `report_trait_object_addition_traits_error` for both